backend/services/unified_samurai_agent.py

```
@@ -1439,63 +1415,214 @@ async def _generate_task_breakdown(self, message: str, context: ConversationCont
+            if cleaned_response.startswith('```'):
+                cleaned_response = cleaned_response[3:]
+            if cleaned_response.endswith('```'):
+                cleaned_response = cleaned_response[:-3]
+            cleaned_response = cleaned_response.strip()
+            
+            # Try direct parsing
+            parsed_response = json.loads(cleaned_response)
+            
+            # Handle special cases from enhanced prompt
+            if isinstance(parsed_response, dict):
+                if parsed_response.get("breakdown_needed") == False:
```

Ruff (0.12.2)

Avoid equality comparisons to `False`; use `not parsed_response.get("breakdown_needed"):` for false checks

Replace with `not parsed_response.get("breakdown_needed")`

(E712)